## In this project we will load iris dataset and from sklearn

In [1]:
from sklearn import datasets

In [2]:
iris=datasets.load_iris()

In [3]:
iris

{'data': array([[5.1, 3.5, 1.4, 0.2],
        [4.9, 3. , 1.4, 0.2],
        [4.7, 3.2, 1.3, 0.2],
        [4.6, 3.1, 1.5, 0.2],
        [5. , 3.6, 1.4, 0.2],
        [5.4, 3.9, 1.7, 0.4],
        [4.6, 3.4, 1.4, 0.3],
        [5. , 3.4, 1.5, 0.2],
        [4.4, 2.9, 1.4, 0.2],
        [4.9, 3.1, 1.5, 0.1],
        [5.4, 3.7, 1.5, 0.2],
        [4.8, 3.4, 1.6, 0.2],
        [4.8, 3. , 1.4, 0.1],
        [4.3, 3. , 1.1, 0.1],
        [5.8, 4. , 1.2, 0.2],
        [5.7, 4.4, 1.5, 0.4],
        [5.4, 3.9, 1.3, 0.4],
        [5.1, 3.5, 1.4, 0.3],
        [5.7, 3.8, 1.7, 0.3],
        [5.1, 3.8, 1.5, 0.3],
        [5.4, 3.4, 1.7, 0.2],
        [5.1, 3.7, 1.5, 0.4],
        [4.6, 3.6, 1. , 0.2],
        [5.1, 3.3, 1.7, 0.5],
        [4.8, 3.4, 1.9, 0.2],
        [5. , 3. , 1.6, 0.2],
        [5. , 3.4, 1.6, 0.4],
        [5.2, 3.5, 1.5, 0.2],
        [5.2, 3.4, 1.4, 0.2],
        [4.7, 3.2, 1.6, 0.2],
        [4.8, 3.1, 1.6, 0.2],
        [5.4, 3.4, 1.5, 0.4],
        [5.2, 4.1, 1.5, 0.1],
  

In [4]:
print(iris.target_names)

['setosa' 'versicolor' 'virginica']


In [5]:
print(iris.feature_names)

['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


## Created new dataset using iris dataset 

In [8]:
import pandas as pd
data= pd.DataFrame({'sepal length':iris.data[:,0],
              'sepal width':iris.data[:,1],
               'petal length':iris.data[:,2],
               'petal width':iris.data[:,3],
                'species':iris.target}) 

In [11]:
data.head()

sepal length  sepal width  petal length  petal width  species
0           5.1          3.5           1.4          0.2        0
1           4.9          3.0           1.4          0.2        0
2           4.7          3.2           1.3          0.2        0
3           4.6          3.1           1.5          0.2        0
4           5.0          3.6           1.4          0.2        0

## Independent & Dependent Variable seperate out here play is only dependent and all others independent

In [12]:
X = data[['sepal length','sepal width','petal length','petal width']]
Y = data['species']

In [13]:
X

sepal length  sepal width  petal length  petal width
0             5.1          3.5           1.4          0.2
1             4.9          3.0           1.4          0.2
2             4.7          3.2           1.3          0.2
3             4.6          3.1           1.5          0.2
4             5.0          3.6           1.4          0.2
..            ...          ...           ...          ...
145           6.7          3.0           5.2          2.3
146           6.3          2.5           5.0          1.9
147           6.5          3.0           5.2          2.0
148           6.2          3.4           5.4          2.3
149           5.9          3.0           5.1          1.8

[150 rows x 4 columns]

In [14]:
Y

0      0
1      0
2      0
3      0
4      0
      ..
145    2
146    2
147    2
148    2
149    2
Name: species, Length: 150, dtype: int32

## Dividing the dataset into training and testing dataset

In [17]:
from sklearn.model_selection import train_test_split

In [18]:
x_train,x_test,y_train,y_test=train_test_split(X,Y,test_size=0.3)

In [19]:
len(x_train)

105

In [20]:
len(x_test)

45

## Random Forest  Classifier Import

In [21]:
from sklearn.ensemble import RandomForestClassifier

In [22]:
classifier=RandomForestClassifier(n_estimators=100,criterion='gini')

## Model training using Fit methods

In [23]:
classifier.fit(x_train,y_train)

RandomForestClassifier()

In [24]:
classifier.predict(x_test)

array([2, 2, 2, 1, 0, 0, 1, 1, 2, 1, 1, 2, 0, 2, 2, 0, 0, 0, 2, 0, 1, 0,
       1, 0, 1, 2, 0, 2, 2, 2, 1, 2, 1, 2, 1, 1, 0, 2, 0, 2, 1, 2, 2, 1,
       2])

In [25]:
x_test  #(0: Setosa,  1: Versicolor,   2: Virginica)

sepal length  sepal width  petal length  petal width
124           6.7          3.3           5.7          2.1
103           6.3          2.9           5.6          1.8
136           6.3          3.4           5.6          2.4
82            5.8          2.7           3.9          1.2
20            5.4          3.4           1.7          0.2
12            4.8          3.0           1.4          0.1
67            5.8          2.7           4.1          1.0
64            5.6          2.9           3.6          1.3
132           6.4          2.8           5.6          2.2
134           6.1          2.6           5.6          1.4
92            5.8          2.6           4.0          1.2
117           7.7          3.8           6.7          2.2
41            4.5          2.3           1.3          0.3
121           5.6          2.8           4.9          2.0
125           7.2          3.2           6.0          1.8
35            5.0          3.2           1.2          0.2
4             5.0          3.6           1.4          0.2
1             4.9          3.0           1.4          0.2
70            5.9          3.2           4.8          1.8
6             4.6          3.4           1.4          0.3
66            5.6          3.0           4.5          1.5
8             4.4          2.9           1.4          0.2
69            5.6          2.5           3.9          1.1
28            5.2          3.4           1.4          0.2
59            5.2          2.7           3.9          1.4
137           6.4          3.1           5.5          1.8
5             5.4          3.9           1.7          0.4
104           6.5          3.0           5.8          2.2
141           6.9          3.1           5.1          2.3
126           6.2          2.8           4.8          1.8
74            6.4          2.9           4.3          1.3
100           6.3          3.3           6.0          2.5
50            7.0          3.2           4.7          1.4
107           7.3          2.9           6.3          1.8
85            6.0          3.4           4.5          1.6
86            6.7          3.1           4.7          1.5
43            5.0          3.5           1.6          0.6
145           6.7          3.0           5.2          2.3
3             4.6          3.1           1.5          0.2
110           6.5          3.2           5.1          2.0
97            6.2          2.9           4.3          1.3
131           7.9          3.8           6.4          2.0
147           6.5          3.0           5.2          2.0
119           6.0          2.2           5.0          1.5
142           5.8          2.7           5.1          1.9

In [26]:
classifier.score(x_test,y_test)

0.9333333333333333

In [27]:
classifier.predict([[3,5,4,2]])

C:\jupyter\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but RandomForestClassifier was fitted with feature names
  warnings.warn(


array([2])

In [28]:
classifier.predict([[3,3,2,2]])

C:\jupyter\lib\site-packages\sklearn\base.py:450: UserWarning: X does not have valid feature names, but RandomForestClassifier was fitted with feature names
  warnings.warn(


array([0])

## To show important features from dataset

In [30]:
Imp_feature=pd.Series(classifier.feature_importances_,index=iris.feature_names).sort_values(ascending=False)

In [31]:
Imp_feature

petal width (cm)     0.514714
petal length (cm)    0.364458
sepal length (cm)    0.082677
sepal width (cm)     0.038151
dtype: float64

## Here we can see top feature for good accuracy is petal width and least feature is sepal width so we will remove it

In [33]:
X=data[['petal length','petal width','sepal length']]
Y=data['species']

In [34]:
x_train,x_test,y_train,y_test=train_test_split(X,Y,test_size=0.3)

In [35]:
clf=RandomForestClassifier()

In [36]:
clf.fit(x_train,y_train)

RandomForestClassifier()

In [37]:
clf.predict(x_test)

array([0, 2, 2, 1, 1, 2, 2, 2, 2, 1, 0, 2, 2, 1, 1, 1, 2, 0, 0, 1, 0, 1,
       1, 0, 1, 0, 0, 0, 0, 0, 0, 2, 0, 1, 2, 0, 0, 2, 1, 1, 2, 1, 1, 0,
       1])

In [39]:
clf.score(x_test,y_test)

0.9777777777777777

## From the above two results we can see our accuracy is improving previously it was 93% and now it  is showing 97%